src/vm/i386/umthunkstub.S

```diff
@@ -173,25 +173,20 @@ LOCAL_LABEL(UMThunkStub_CopyStackArgs):
 LOCAL_LABEL(InitCopyStack):
     push    ecx
     push    edx
-    mov     eax, 0  // Offset in edi
-    mov     ebx, 0  // Offset in esi
-LOCAL_LABEL(CopyStack):
     mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
     mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
-    mov     ecx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
-    cmp     ecx, ebx
+    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
```

Comment:
> The CF is set by the add ebx, -4 before the conditional jump. 

You are right, I missed that. `jc` works fine, thanks a lot!
BTW, 2 tests (Vector3Interop_ro.sh and Vector3Interop_r.sh) failed on x64 checks, so I should investigate the reason of this regression.